helloworld/notebooks/OpHousingPrices.ipynb

```diff
@@ -0,0 +1,394 @@
+    "and are strongly typed. TransmogrifAI supports the following basic feature types: `Text`, \n",
+    "`Numeric`, `Vector`, `List` , `Set`, `Map`. In addition it supports many specific feature \n",
+    "types which extend these base types: Email extends Text; Integral, Real and Binary extend \n",
+    "Numeric; Currency and Percentage extend Real. For a complete view of the types supported \n",
+    "see the Type Hierarchy and Automatic Feature Engineering section in the Documentation.\n",
+    "\n",
+    "Basic `FeatureBuilders` will be created for you if you use the TransmogrifAI CLI to bootstrap \n",
+    "your project as described here. However, it is often useful to edit this code to customize \n",
+    "feature generation and take full advantage of the Feature types available (selecting the \n",
+    "appropriate type will improve automatic feature engineering steps).\n",
+    "When defining raw features, specify the extract logic to be applied to the raw data, and \n",
+    "also annotate the features as either predictor or response variables via the FeatureBuilders:\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import org.apache.spark.sql.{Encoders}\n",
+    "implicit val srEncoder = Encoders.product[HousingPrices]\n",
+    "val saleTypeEncoder = Map(\"COD\" -> 1, \"CWD\" -> 2, \"Con\" -> 3, \"ConLD\" -> 4,\n",
+    "      \"ConLI\" -> 5, \"ConLw\" -> 6,\"New\" -> 7, \"Oth\" -> 8,\"WD\" -> 9  )\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "val lotFrontage = FeatureBuilder.Real[HousingPrices].extract(_.lotFrontage.toReal).asPredictor\n",
+    "val area = FeatureBuilder.Integral[HousingPrices].extract(_.area.toIntegral).asPredictor"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "val lotShape = FeatureBuilder.Integral[HousingPrices].extract(_.lotShape match {\n",
+    "    case \"IR1\" => 1.toIntegral\n",
+    "    case _ => 0.toIntegral\n",
+    "}).asPredictor"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "val yrSold = FeatureBuilder.Integral[HousingPrices].extract(x => (2019 - x.yrSold).toIntegral).asPredictor"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "val saleType = FeatureBuilder.Integral[HousingPrices].extract(x => saleTypeEncoder.get(x.saleType).toIntegral).asPredictor"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "val saleConditionEncoder = Map(\"Abnorml\" -> 1, \"AdjLand\" -> 2, \"Alloca\" -> 3, \"Family\" -> 4,\n",
```

Comment:
same here why are you converting to int? this is a picklist.

Reply:
if they work better in the model as encoded variable can you use the string indexer before calling .transmografai()

https://github.com/salesforce/TransmogrifAI/blob/master/core/src/test/scala/com/salesforce/op/stages/impl/feature/OpStringIndexerNoFilterTest.scala#L75

"val saleTypeInd = saleType.indexed()"
val saleCondInd = saleCondition.indexed()"
"val features = Seq(lotFrontage,area,lotShape, yrSold, saleTypeInd, saleCondInd).transmogrify()\n",


This is preferred because model insights will have the correct names if you use our internal indexer

Reply:
With encoded variables i am not able to use indexed()

```
<console>:149: error: value indexed is not a member of com.salesforce.op.features.Feature[com.salesforce.op.features.types.Integral]
       val saleCondition = FeatureBuilder.Integral[HousingPrices].extract(x => saleConditionEncoder.get(x.saleCondition).toIntegral).asPredictor.indexed()
       
```

Reply:
The indexer is for strings - put the original string in - not the mapped integer value. The indexer will then do the mapping for you. But unlike in the map you are creating the name to index mapping will be saved in the feature metadata so any feature insights will have the name associated with it. It will also not treat it as an ordered integer in modeling which would be inappropriate

Reply:
changed